src/contract_wrappers/tokens/erc20_wrapper.ts

```diff
@@ -159,10 +160,27 @@ export default abstract class ERC20TokenWrapper extends ContractWrapper {
   public async isValidContract() {
     try {
       const contract = await this.contract;
-      await contract.totalSupply.callAsync();
-      await contract.balanceOf.callAsync(contract.address);
-      await contract.allowance.callAsync(contract.address, contract.address);
-      await contract.symbol.callAsync();
+      const totalSupply = await contract.totalSupply.callAsync();
+      const symbol = await contract.symbol.callAsync();
+      const name = await contract.name.callAsync();
+      if (symbol === '' || name === '' || totalSupply.isZero()) {
+        return false;
+      }
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  public async isValidAlternativeContract() {
```

Comment:
You could move this to AlternativeERC20TokenWrapper overriding the original implementation, so from outside you are only calling to `isValidContract()'